sdk/core/Azure.Core.TestFramework/src/ClientTestBase.cs

```diff
@@ -21,13 +24,75 @@ public abstract class ClientTestBase
         private static readonly IInterceptor s_diagnosticScopeValidatingInterceptor = new DiagnosticScopeValidatingInterceptor();
         private static Dictionary<Type, Exception> s_clientValidation = new Dictionary<Type, Exception>();
         public bool IsAsync { get; }
-
+        protected static TestLogger Logger { get; set; }
         public bool TestDiagnostics { get; set; } = true;
+        public RecordedTestMode Mode { get; set; }
+        public TestRecording Recording { get;  set; }
+
+        // copied the Windows version https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs
+        // as it is the most restrictive of all platforms
+        private static readonly HashSet<char> s_invalidChars = new HashSet<char>(new char[]
+        {
+            '\"', '<', '>', '|', '\0',
+            (char)1, (char)2, (char)3, (char)4, (char)5, (char)6, (char)7, (char)8, (char)9, (char)10,
+            (char)11, (char)12, (char)13, (char)14, (char)15, (char)16, (char)17, (char)18, (char)19, (char)20,
+            (char)21, (char)22, (char)23, (char)24, (char)25, (char)26, (char)27, (char)28, (char)29, (char)30,
+            (char)31, ':', '*', '?', '\\', '/'
+        });
+        protected string GetSessionFilePath()
```

Comment:
This is a copy of logic from recorded test base. It doesn't belong in this file.

Reply:
It's been removed from recorded test base, since functionality used in ManagmentRecordTestBase and RecordedTestBase and we don't want to Inherit from RecrodedTestBase

Reply:
>we don't want to Inherit from RecrodedTestBase

Why?

Reply:
From above:

Because we don't want to touch this file since we want it as a option to run the test as they currently are. Also, we don't want these [OneTimeSetup] [Setup] [OneTimeTearDown] [TearDown] to run in that file. We want to control those functions/methods with those tags.

Reply:
Not sure this directly has anything to do with "recordings".  It is trying to find the proper location to store / retrieve anything that is needed.  Perhaps a Mocked test will store the mocked responses in files and can use the exact same location with an already established common pattern where local and ci can find it easily.  How a session is created has nothing to do with "recording" a test per se, that is just one way to do it.

Reply:
>Not sure this directly has anything to do with "recordings".

Storing recordings is the only thing it's used for.

>Perhaps a Mocked test will store the mocked responses in files and can use the exact same location with an already established common pattern where local and ci can find it easily

This is not the case, none of the tests is using the SessionRecords directory for anything but test recordings and we should keep it that way.